packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -420,7 +420,19 @@ - (int64_t)duration {
 }
 
 - (void)seekTo:(int)location completionHandler:(void (^)(BOOL))completionHandler {
-  [_player seekToTime:CMTimeMake(location, 1000)
+  CMTime locationCMT = CMTimeMake(location, 1000);
+  CMTimeValue duration = _player.currentItem.asset.duration.value;
+  CMTimeValue locationCM = locationCMT.value;
+  if (duration == locationCM) {
+    // Without adding tolerance when seeking to duration,
+    // seekToTime will never complete, and this call will hang.
+    [_player seekToTime:_player.currentItem.currentTime
+          toleranceBefore:CMTimeMake(1, 1000)
+           toleranceAfter:CMTimeMake(1, 1000)
+        completionHandler:completionHandler];
+    return;
+  }
```

Comment:
We generally want to minimize the amount of code in a branch that is mostly identical, since we can get accidental drift over time. This can instead be:

```
// your explanatory comment here.
CMTime tolerance = locationCM == duration ? CMTimeMake(1, 1000) : kCMTimeZero;
[_player seekToTime:locationCMT
        toleranceBefore:tolerance
         toleranceAfter:tolerance
      completionHandler:completionHandler];
```

(And in fact while writing that I noticed that your `if` branch was seeking to the current time, which means this would have regressed seeking to the end in every case other than the one you were debugging, so the duplication was already problematic.)

Reply:
Actually, hrm. How was your integration test passing with that?

Reply:
This doesn't prevent seeking to the end. In practice, it shouldn't actually do anything. It just allows a small tolerance in accuracy when seeking to the end.